### PR TITLE
[Mac] Cleanup MONO_REGISTRY_PATH on startup

### PR DESCRIPTION
**Pull request by rodrmoya**
If started from a Xamarin.Mac app, this env var is set pointing to a (maybe)
non-existing folder (~/Library/Application Support/Whatever), which causes
code that tries to access the registry fail if that folder doesn't exist. So,
since we want to use Mono's registry, make sure the env var is cleaned up
on startup.

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/896438

### CODE REVIEW

**Comment by rodrmoya** [comment]
@monojenkins backport release-8.1

File: main/build/MacOSX/monostub.mm [hunk 1]
```diff
@@ -269,6 +269,7 @@
 				req_mono_version = version_obj;
 		}
 
+		unsetenv ("MONO_REGISTRY_PATH");
```
Comment (slluis) [inline main/build/MacOSX/monostub.mm]: Could you add a comment explaining why was this added?

Comment (rodrmoya) [reply main/build/MacOSX/monostub.mm]: Done!

**Comment by rodrmoya** [comment]
Backported manually at https://github.com/mono/monodevelop/pull/7660
